sdk/core/Azure.Core/src/Pipeline/HttpPipeline.cs

```diff
@@ -144,6 +212,51 @@ public static IDisposable CreateHttpMessagePropertiesScope(IDictionary<string, o
             return CurrentHttpMessagePropertiesScope.Value;
         }
 
+        private ReadOnlyMemory<HttpPipelinePolicy> CreateRequestPipeline(HttpPipelinePolicy[] policies, List<(HttpPipelinePosition Position, HttpPipelinePolicy Policy)> customPolicies)
+        {
+            if (!_internallyConstructed)
+            {
+                throw new InvalidOperationException("Cannot send messages with per-request policies if the pipeline wasn't constructed with HttpPipelineBuilder.");
+            }
+
+            int AddCustomPolicies(HttpPipelinePosition position, int start)
+            {
+                int i = 0;
+                if (customPolicies != null)
+                {
+                    foreach (var policy in customPolicies)
+                    {
+                        if (policy.Position == position)
+                        {
+                            policies[start + i++] = policy.Policy;
```

Comment:
![https://media3.giphy.com/media/3owzW5c1tPq63MPmWk/giphy.gif](https://media3.giphy.com/media/3owzW5c1tPq63MPmWk/giphy.gif)

Let's try and simplify this expression a bit :)

Reply:
I made it go away.